src/mapgen/dungeongen.cpp

```diff
@@ -219,7 +212,7 @@ void DungeonGen::makeDungeon(v3s16 start_padding)
 #endif
 
 		// Quit if last room
-		if (i == room_count - 1)
+		if (i + 1 == dp.num_rooms)
```

Comment:
The `-1` is moved to the other side to become `+1` to avoid the right side being changed to a signed int and triggering a 'comparison between signed and unsigned int' build warning.